pyanaconda/geoloc.py

```diff
@@ -192,8 +193,9 @@ def _check_if_geolocation_should_be_used(self, options_override):
                 geolocation_enabled = False
 
         # and also check if geolocation was not disabled by boot or command like option
-        if not flags.cmdline.getbool('geoloc', True):
-            # flags.cmdline.getbool is used as it handles values such as
+        if not kernel_arguments.get('geoloc', default=True, interpret_booleans=True) or \
```

Comment:
I would still prefer to call `kernel_arguments.get('geoloc', default=True, converter=bool)`, because it is a more universal solution. For example, we would be able to call `kernel_arguments.get('timeout', default=0, converter=int)` to retrieve a number from the kernel arguments. However, the current solution is all right, so I don't mind.

Reply:
I think it could be even more robust by finding bugs when the value is not a given type.

Reply:
I don't really agree. In hindsight, I should have argued this earlier. The core of my argument is that I don't like making the interface look as if it can just take any callable, then unexpected magic springs from around the corner, making booleans behave weirdly, when the syntax is the same as for any other callables.

The legacy rules for parsing "bools" are "no/off/0" || not present VS any other case. Let's compare with the other place you suggested which is based on `configparser.RawConfigParser.BOOLEAN_STATES`: "false" is *not* going to give False, and presence is interpreted too. So these are not consistent. 

I have no other word other than *magical* for the behavior that would result from making `bool` get the magic in place of a callable. The method signature implies a `bool` that could be interchangeable with `int`, except it does something completely unexpected. So I don't want to make it look like you can just plug in anything with equal results and rules, because you can *not*.

If I were to go to the full extent of my argument, then even adding the parameter there (as I eventually did) makes no sense - containing the magic is best done by letting it have its own clearly separate method. It's not like the calls do not have to be different anyway:  `get(..., interpret_booleans=True)` != `getbool(...)` != `get(..., converter=bool)`. People will have to type different things anyway, and the legacy logic for the magic must be retained too, so we are not getting better in any metric.

Perhaps that's too subtle, but it is also the reason I went with "interpret" instead of "convert" in the parameter name as it (to me) implies more processing.

As to callables. The only advantage allowed by using a callable parameter is that one could add their own conversion later, but one can do `get(...,converter=int)` equally well as just `int(get(...))`. So why? Callable is a callable, whether in or out of the method signature. And making it so for a mere *possibility* of somebody using a converter in the future seems a premature optimization.

...so yes, I am convinced the converter argument in the other place is not good either.

I don't want to make it look like I care too much about this - the previous state is suboptimal, I don't see how the next one could be much better, so if you think defaults+callables+magic in single method is the way to go, I am not saying *never!* It's just, I don't see any argument for doing it that way. So I expect a convincing argument "why".

Reply:
@jkonecny12 @M4rtinK @rvykydal - After offline discussion with @poncovka, we decided the way forward is: Remove `interpret_booleans` parameter, and use a new method `is_enabled(str)` instead. We'll also drop the yes/no and keep only 0/1 & on/off.